### PR TITLE
Fix memory usage and add update command.

### PR DESCRIPTION
**Pull request by fennecdjay**
This PR improves memory usage and add a (simple) update command.
I tried to make each commit as simple as possible, yet it might not be perfect.
Thanks for the work.
let me know.

### CODE REVIEW

**Comment by desyncr** [comment]
You are awesome. Thank you very much.

I'm gonna take a look tonight.

File: zpm.c [hunk 1]
```diff
@@ -227,13 +249,31 @@ char* get_zpm_plugin_list() {
...
     return listing;
 }
 
 int plugins_update_local_clone() {
+    int ret;
+    char* command = malloc(PATH_MAX);
+    char* listing = get_zpm_plugin_list();
+    char* plugin_name = strtok(listing, "\n");
+
+strcpy(command, "echo lol ");
```
Comment (desyncr) [inline zpm.c]: Remove this.

File: zpm.c [hunk 2]
```diff
@@ -227,13 +249,31 @@ char* get_zpm_plugin_list() {
...
+
```
Comment (desyncr) [inline zpm.c]: Add a "Updating plugins..." message around here.

**Review by desyncr — CHANGES_REQUESTED** [review]
Looks good to me! Thanks a lot.

There is these two small things and we'll be ready to merge.

File: zpm.c (hunk 1 above)
Comment (fennecdjay) [reply zpm.c]: sorry about this one.

**Comment by fennecdjay** [comment]
Great.
I give a try.

**Comment by fennecdjay** [comment]
This might be ok now.

**Comment by desyncr** [comment]
Cool thanks. Anyways from my side it doesn't seem to work as expected. It seems to only iterate over the first plugin listed, rather than over all of them.

**Comment by fennecdjay** [comment]
My mistake.
my config was back to only one plugin, so I missed that.
I'll fix fix ASAP.

**Comment by desyncr** [comment]
Don't worry, great work so far!

**Comment by fennecdjay** [comment]
Now system is called for every installed plugin.
Also I updated the README to reflect current state.

File: zpm.c [hunk 3]
```diff
@@ -227,13 +249,28 @@ char* get_zpm_plugin_list() {
...
+    char* plugin_name = strtok(listing, "\n");
+
+    while(plugin_name) {
+      strcpy(command, "cd ~/.zpm/plugins/");
+      strcat(command, plugin_name);
+      strcat(command, "; git pull");
+      ret = system(command);
+      plugin_name = strtok(NULL, "\n");
+    }
+    printf("Update plugins ...\n");
```
Comment (desyncr) [inline zpm.c]: This `printf` should be before the `while` loop, not after it. Be sure it reads "Updating plugins...".

File: zpm.c [hunk 4]
```diff
@@ -227,13 +249,28 @@ char* get_zpm_plugin_list() {
...
-    // iterate over plugins_list
-    // exec git pull --git-dir=
-    return 0;
+    int ret;
+    char* command = malloc(PATH_MAX);
+    char* listing = get_zpm_plugin_list();
+    char* plugin_name = strtok(listing, "\n");
+
+    while(plugin_name) {
+      strcpy(command, "cd ~/.zpm/plugins/");
```
Comment (desyncr) [inline zpm.c:269]: Maybe we could add a `"Updating %s...", plugin_name`?

**Review by desyncr — CHANGES_REQUESTED** [review]
We're getting there :+1: A few small changes.

**Comment by desyncr** [comment]
Looking forward to merge this! Great work!

**Comment by fennecdjay** [comment]
This is better. I also fixed a bug when list is empty.
Let me know I you want individual plugin update message.

File: zpm.c [hunk 5]
```diff
@@ -227,13 +249,32 @@ char* get_zpm_plugin_list() {
...
+
+    if(!strcmp(listing, "Nothing to show.")) {
+      printf("Nothing to update.");
+      return -1;
+    }
+    printf("Updating plugins ...\n");
+    while(plugin_name) {
+      strcpy(command, "cd ~/.zpm/plugins/");
+      strcat(command, plugin_name);
+      strcat(command, "; git pull");
```
Comment (desyncr) [inline zpm.c:271]: Adding a print below this line would be the final touch up:

    printf("Updating %s...\n", plugin_name);

**Review by desyncr — CHANGES_REQUESTED** [review]
One final touch up and we're done! :D

File: zpm.c (hunk 5 above)
Comment (fennecdjay) [reply zpm.c:271]: Done!!!

Comment (desyncr) [reply zpm.c:271]: Awesome! Gonna merge now!

**Pull request merged**